src/lib/Underground.js

```diff
@@ -191,6 +202,12 @@ class AutomationUnderground
             return;
         }
 
+        // Check for daily deals, if the player enabled the feature
+        if (Automation.Utils.LocalStorage.getValue(this.Settings.TradeDiamonds) === "true")
+        {
+            this.__internal__tradeDiamonds();
+        }
+
```

Comment:
You moved this to AFTER the check for the innerMiningLoop, which will make it not work for some cases.

For example, I have A LOT of restore items, and spend all my money in more restore items. Until the restore items run out (and they might NOT run out, or take a lot of time), the check for the diamond trades will not run, as the inner loop is still running. This will make it miss some trades.

I would like to move it back before the check for the inner loop, but only if you agree

Reply:
Ignore this comment, it is in the correct place and I misunderstood which function was this one.

Reply:
I indeed changed it's place already ;)

Reply:
Yeah, I moved it, because this check implies that the user disabled the feature if it's null
So we do not want it to do any more processing

Reply:
Ignore the coment to ignore the previous comment.

This code either has to go before the check for `if (this.__internal__innerMiningLoop !== null)` or inside the `this.__internal__innerMiningLoop = setInterval(function()` loop itself.

Right now, it is not running until the player runs out of energy

I will move it back before the check for innerMiningLoop so it runs every 10 seconds and not every 0.3 seconds.

Tagging in case the previously resolved conversation makes the notification not get pushed @Farigh

Reply:
Hum, in such case, it should have it's own interval, which should run every hour for example

Reply:
Maybe every one to five minutes, but I was simply piggybacking off the already existing 10 second interval.

I simply did not want to bother having to define a new interval.

If I create the interval in `toggleAutoMining()` it should be handled correctly, right?